pkg/planner/core/plan_cache.go

```diff
@@ -116,7 +116,7 @@ func planCachePreprocess(ctx context.Context, sctx sessionctx.Context, isNonPrep
 			schemaNotMatch = true
 			continue
 		}
-		if stmt.tbls[i].Meta().Revision != newTbl.Meta().Revision {
+		if stmt.tbls[i].Meta().Revision != newTbl.Meta().Revision || (tbl != nil && tbl.Meta().Revision != newTbl.Meta().Revision) {
```

Comment:
IMO, add some comments about why not only `stmt.tbls` but also `tbl` need to be check is better

Reply:
This scene is quite difficult to describe, you see my PR description written quite a lot. Do you have any suggestions?